src/nupic/research/connections.py

```diff
@@ -20,23 +20,133 @@
 # ----------------------------------------------------------------------
 
 from collections import defaultdict
+from bisect import bisect_left
 
+EPSILON = 0.00001 # constant error threshold to check equality of permanences to
+                  # other floats
 
 
-class SynapseData(object):
+class Segment(object):
+  ''' Class containing minimal information to identify a unique segment '''
```

Comment:
nit: Some of these docstrings are using single quotes, others are using double quotes. I think they should all be double.